cpp/build-support/asan_symbolize.py

```diff
@@ -318,15 +321,20 @@ def symbolize_address(self, addr, binary, offset):
 
   def print_symbolized_lines(self, symbolized_lines):
     if not symbolized_lines:
-      print self.current_line
+      print(self.current_line)
     else:
       for symbolized_frame in symbolized_lines:
-        print '    #' + str(self.frame_no) + ' ' + symbolized_frame.rstrip()
+        print('    #' + str(self.frame_no) + ' ' + symbolized_frame.rstrip())
         self.frame_no += 1
 
   def process_stdin(self):
     self.frame_no = 0
-    sys.stdout = os.fdopen(sys.stdout.fileno(), 'w', 0)
+
+    if sys.version_info[0] == 2:
+      sys.stdout = os.fdopen(sys.stdout.fileno(), 'w', 0)
+    else:
+      # Unbuffered output is not supported in Python 3
+      sys.stdout = os.fdopen(sys.stdout.fileno(), 'w')
```

Comment:
Looking at the code I see no benefit except that it may limit the memory usage for the buffers in case of large stacktraces that can not be processed fast enough by the process consuming `sys.stdout`.